x11-dl/src/link.rs

```diff
@@ -34,31 +34,34 @@ macro_rules! x11_link {
     unsafe impl Sync for $struct_name {}
 
     impl $struct_name {
-      unsafe fn init (this: *mut Self) -> Result<(), $crate::error::OpenError> {
-        lazy_static! {
-          static ref SYMS: [(&'static str, usize); $nsyms] = unsafe {[
-            $((stringify!($fn_name), &((*core::ptr::null::<$struct_name>()).$fn_name) as *const _ as usize),)*
-            $((stringify!($vfn_name), &((*core::ptr::null::<$struct_name>()).$vfn_name) as *const _ as usize),)*
-            $((stringify!($var_name), &((*core::ptr::null::<$struct_name>()).$var_name) as *const _ as usize),)*
```

Comment:
Also FYI, the old code that was removed here has Undefined Behavior -- offsetting a null pointer by a non-zero amount (such as to project to the field, as in this code) is not allowed. Lucky enough we have `offset_of!` these days which makes it much less likely that people write code like this. :)